apps/explorer/src/app/components/txs/tx-order-type.tsx

```diff
@@ -36,6 +37,31 @@ const displayString: StringMap = {
   ValidatorHeartbeat: 'Validator Heartbeat',
 };
 
+/**
+ * Given a proposal, will return a specific label
+ * @param chainEvent
+ * @returns
+ */
+export function getLabelForProposal(
+  proposal: components['schemas']['v1ProposalSubmission']
+): string {
+  if (proposal.terms?.newAsset) {
```

Comment:
comment (non-blocking): as a real sucker for `switch`, i'd like to take the opportunity to promote my favourite, 🥁  the switch statement! it's real and it's free, not a pyramid scheme! (couldn't make it rhyme properly 😢 )

Reply:
yeah, actually as pointed out by you in the chat, the if checks are targeting different properties of the same object. false marketing from my side.

Reply:
Thanks for coming back here. I asked directly because I thought you were going to pull out some ingenious switch usage that I had missed. I'm sure it's possible 😄